### PR TITLE
[SYCL] Change ESIMD/aot_mixed.cpp to AOT-compile for all devices

### PR DESCRIPTION
**Pull request by aelovikov-intel**
This should fix the CI fails on it.

### CODE REVIEW

**Review by v-klochkov — COMMENTED** [review]
I did not see '-device *' in the list of options of 'ocloc'.
@vmustya suggested using "device gen9:" (see ":" symbol after 9), which means "gen9 and newer".

**Comment by aelovikov-intel** [comment]
I couldn't find any documentation for it at all. That said:

```
$ ocloc -file a.cl -device '*'
Build succeeded for : bdw.
Build succeeded for : skl.
Build succeeded for : kbl.
Build succeeded for : cfl.
Build succeeded for : apl.
Build succeeded for : glk.
Build succeeded for : whl.
Build succeeded for : aml.
Build succeeded for : cml.
Build succeeded for : icllp.
Build succeeded for : lkf.
Build succeeded for : ehl.
Build succeeded for : tgllp.
Build succeeded for : rkl.
Build succeeded for : adl-s.
Build succeeded for : adl-p.
Build succeeded for : adl-n.
Build succeeded for : dg1.
Build succeeded for : acm-g10.
Build succeeded for : acm-g11.
Build succeeded for : acm-g12.
Build succeeded for : pvc-sdv.
Build succeeded for : pvc.
```

From https://www.phoronix.com/news/Intel-Compute-20.07.15711:

> There is also -device * now supported if wanting to compile all hardware targets supported by the Intel compiler.

I'm sure I had found the proper docs in the past, but it seems impossible now...

**Comment by v-klochkov** [comment]
'device *' doesn't work accordingly to our CI: http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/LLVM-Test-Suite-CI-TMP%2FLLVM-Test-Suite-CI-Linux/detail/LLVM-Test-Suite-CI-Linux/3241/pipeline
The test fails at ocloc compilation " LLVM ERROR: vc::translateBuild: unsupported platform"

Please try "-device gen9:"

File: SYCL/ESIMD/aot_mixed.cpp [hunk 1]
```diff
@@ -8,9 +8,9 @@
 // REQUIRES: gpu
 // UNSUPPORTED: cuda || hip
 // UNSUPPORTED: esimd_emulator
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" -o %t.sycl.out -DENABLE_SYCL=0 %s
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device *" -o %t.sycl.out -DENABLE_SYCL=0 %s
```
Comment (aelovikov-intel) [inline SYCL/ESIMD/aot_mixed.cpp]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9:" -o %t.sycl.out -DENABLE_SYCL=0 %s
```

File: SYCL/ESIMD/aot_mixed.cpp [hunk 2]
```diff
@@ -8,9 +8,9 @@
...
 // RUN: %GPU_RUN_PLACEHOLDER %t.sycl.out
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" -o %t.out %s
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device *" -o %t.out %s
```
Comment (aelovikov-intel) [inline SYCL/ESIMD/aot_mixed.cpp]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9:" -o %t.out %s
```

File: SYCL/ESIMD/aot_mixed.cpp [hunk 3]
```diff
@@ -8,9 +8,9 @@
 // REQUIRES: gpu
 // UNSUPPORTED: cuda || hip
 // UNSUPPORTED: esimd_emulator
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" -o %t.sycl.out -DENABLE_SYCL=0 %s
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9:" -o %t.sycl.out -DENABLE_SYCL=0 %s
```
Comment (aelovikov-intel) [inline SYCL/ESIMD/aot_mixed.cpp:11]:
```suggestion
// Ideally, we should be using "-device *" but there is a bug with backend compiler
// crashing with some of the platforms. Try to cover our CI pool for now.
// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9,gen11,tgllp" -o %t.sycl.out -DENABLE_SYCL=0 %s
```

File: SYCL/ESIMD/aot_mixed.cpp [hunk 4]
```diff
@@ -8,9 +8,9 @@
...
 // RUN: %GPU_RUN_PLACEHOLDER %t.sycl.out
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" -o %t.out %s
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9:" -o %t.out %s
```
Comment (aelovikov-intel) [inline SYCL/ESIMD/aot_mixed.cpp:13]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9,gen11,tgllp" -o %t.out %s
```

**Pull request closed without merging**